### ISSUE TITLE
Review Test Parallelism Decorators

### ISSUE THREAD

**Issue by jsquire** [Azure/azure-sdk-for-net#7095]
# Summary

Currently, metadata regarding the parallel approach for tests is applied at the class level of each test suite, either allowing tests to run in parallel or opting out.  This was initially done to allow fine grained control during iterations to better understand the behavior of NUnit's test runner and different components of the Event Hubs client library.

As both of these are better understood now, running tests in parallel should be hoisted to the assembly-level and become the default for tests, allowing them to opt-out individually for test suites or individual tests where parallelism is not desired.  This will reduce clutter in the test classes and avoid the mistake of not running in parallel if an author forgets to apply the attribute.

# Scope of Work

- Add metadata in the assembly properties to allow test suites and tests (`ParallelScope.All`) to run in parallel by default.

- Remove the `Parallelizable` attribute from individual test suites and tests where currently decorated with `[Parallelizable(ParallelScope.All)]`

- Ensure that any test suites or tests decorated with a different parallel scope or `[NonParallelizable]` remain untouched.

# Out of Scope

- Updates to the tests themselves, or addition of new tests.

# Success Criteria

- The unit and live tests continue to pass at the expected rate.  _(At the time of writing, there are 5 skipped tests and a 100% expected pass rate.)_

- The unit and live tests continue to run near the current duration.   _(At the time of writing, the entire set, including live tests, takes about 7 minutes per target platform.)_
